.github/CODE_STANDARDS.md

```diff
@@ -0,0 +1,295 @@
+
+5. **Performance**: Review the code for any potential performance issues, such as inefficient loops, unnecessary allocations, or misuse of concurrency.
+
+6. **Dependency Management**: Ensure that any new dependencies are necessary, appropriately versioned, and have been vetted for performance and security.
+
+7. **Security Practices**: Scrutinize the code for any security vulnerabilities. Validate input, avoid security anti-patterns, and utilize secure coding practices.
+
+8. **Documentation**: Confirm that the code is well-documented, including comments, function/method descriptions, and module-level documentation as necessary.
+
+9. **Efficiency and Readability**: The code should be efficient and readable. Reviewers should look for any code smells, overly complex functions, and ensure the use of idiomatic Go patterns.
+
+10. **Responsiveness**: Both authors and reviewers should be timely in their responses. Authors should address all review comments, and reviewers should re-review changes promptly.
+
+#### 2.3.3 Code Review Checklist
+
+- [ ] Does the code adhere to the project’s coding standards?
+- [ ] Are there sufficient tests, and do they cover a variety of cases?
+- [ ] Is error handling comprehensive and user-friendly?
+- [ ] Are there any performance concerns in the code?
+- [ ] Have new dependencies been appropriately vetted?
+- [ ] Does the code follow secure coding practices and avoid common vulnerabilities?
+- [ ] Is the code well(self)-documented, with clear variable, function naming?
+- [ ] Is the code efficient, readable, and free of code smells?
+- [ ] Have all review comments been addressed in a timely manner?
+- [ ] Do you understand the code and it's purpose?
+
+This checklist serves as a guide to both authors and reviewers to ensure a thorough and effective code review process.
+
+## 3. Contributing
+
+### 3.1 Pull Requests && Issues
+
+We have separate templates for Pull Requests and Issues. Please use them when creating a new PR or Issue.
+
+### 3.2 Conventional Commits & Pull Requests Naming
+
+#### 3.2.1 Overview
+
+In an effort to maintain clarity and coherence in our commit history, we are adopting the Conventional Commits style for all commit messages across our repositories. This uniform format not only enhances the readability of our commit history but also facilitates automated tools in generating changelogs and extracting valuable information effectively.
+
+#### 3.2.2 Structure
+
+Conventional Commits follow a structured format: `type(scope): description`, where:
+
+- `type`: Represents the nature of the commit (e.g., feat, fix, chore).
+- `scope`: Denotes the relevant module or issue.
+- `description`: Provides a brief explanation of the change.
+
+When introducing breaking changes, an `!` should be appended after the `type/scope`: `feat(#123)!: introduce a breaking change`.
+
```

Comment:
Since we want to introduce automatic versioning based on conventional commits, maybe we can explain that here as well, what commit will trigger what version update:

`feat(#123)!: introduce breaking change` - `1.0.0` -> `2.0.0`
`feat(#124): introduce new feature` - `2.0.0` -> `2.1.0`
`fix(#125): fix a bug` - `2.1.0` -> `2.1.1`

Or, if we don't have automatic version updating, we can explain with this example to what version should someone bump when pushing a feature.

Reply:
Good point!

Reply:
Added automatic versioning section :)